### ISSUE TITLE
Add YOLO v7

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#1179]
**Short Description**

> YOLOv7 surpasses all known object detectors in both speed and accuracy in the range from 5 FPS to 160 FPS and has the highest accuracy 56.8% AP among all known real-time object detectors with 30 FPS or higher on GPU V100. YOLOv7-E6 object detector (56 FPS V100, 55.9%
AP) outperforms both transformer-based detector SWINL Cascade-Mask R-CNN (9.2 FPS A100, 53.9% AP) by 509% in speed and 2% in accuracy, and convolutionalbased detector ConvNeXt-XL Cascade-Mask R-CNN (8.6 FPS A100, 55.2% AP) by 551% in speed and 0.7% AP in accuracy, as well as YOLOv7 outperforms: YOLOR, YOLOX, Scaled-YOLOv4, YOLOv5, DETR, Deformable DETR, DINO-5scale-R50, ViT-Adapter-B and many other object detectors in speed and accuracy. Moreover, we train YOLOv7 only on MS COCO dataset from scratch without using any other datasets or pre-trained weights.

**Papers**

- https://arxiv.org/abs/2207.02696

**Existing Implementations**

- Official (pytorch) : https://github.com/WongKinYiu/yolov7
- **tensorflow (keras)**: https://www.kaggle.com/code/jobayerhossain/yolov7-explanation-and-implementation-from-scratch

**Other Information**

As a starter code example now exist, I think it would be easy to add it.

**Comment by tanzhenyu** [comment]
Opening up this for contribution. This is a highly desired feature. We also have v3 & X in progress

**Comment by MrinalTyagi** [comment]
interested in working on this.

**Comment by tanzhenyu** [comment]
> interested in working on this.

Go ahead!

**Comment by innat** [comment]
YOL0-V8 https://v8docs.ultralytics.com/

**Comment by MrinalTyagi** [comment]
@tanzhenyu @innat Sorry for not updating here. I am working towards the addition of the model and will try to share a draft pr by end of the week. Would really appreciate your support.

**Comment by tanzhenyu** [comment]
> YOL0-V8 https://v8docs.ultralytics.com/

Let's open another request to support this, with porting the weights first (while the paper is not officially released so we don't know how it is trained)

**Comment by bhack** [comment]
@MrinalTyagi To have a working devenv you can just click this button:
![immagine](https://user-images.githubusercontent.com/1710528/212680302-171ffc82-5437-4937-ab34-51ac5f9876f8.png)

**Comment by jere357** [comment]
yolov7 in my experience is overfitted to coco possibly a meme paper, me and my colleagues had better results with the ultralytics yolo implementation

**Comment by innat** [comment]
@jere357 what do you mean by overfit and meme paper? Also, ultralytics only provides v3,5,8, doesn't it?

**Comment by jbischof** [comment]
BTW, we are currently working on YOLOv8, so maybe this is less relevant. #1711

**Comment by innat** [comment]
@jbischof 
I think to get attention from ml-practitioners, maybe it's needed to support SOTA model for the most known categories (classification, detection, segmentation, generative ai). In that sense, yes, if yolo-v8 is ready to use, then yolo-vn (n < 8) can be revisited later.  Though contribuor should be welcomed in any time.

(PS. sota for generatice ai, [control-net](https://github.com/keras-team/keras-cv/issues/1456). For egmentation [sam](https://github.com/keras-team/keras-cv/issues/1679). )

**Comment by jere357** [comment]
> @jere357 what do you mean by overfit and meme paper? Also, ultralytics only provides v3,5,8, doesn't it?  


uhh, I don't want to sound like a playa hater but have you read the yolov7 paper? I am a never published disgruntled student of computer science so take that into account when reading this. They talk about outperforming all modern architectures and stuff but in the graph they only compare themselves to 2+ year old papers  mAp on coco is such a bad metric for object detection state-of-the-art", especially when they "Moreover, we train YOLOv7 only on MS COCO dataset from scratch without using any other datasets or pre-trained weights". To me that sounds like literally we ran X experiments with hopes of one of them having a slightly higher mAP in these very controlled circumstances (0 talks about how the network are initialized is hillarious to me) Table 2 results with regards to number of parameters just not impressive at all.

like HAHAHAHHA look at them in chapter 5.3
If we compare YOLOv7-tiny-SiLU with YOLOv5-N (r6.1), our method is 127 fps faster and 10.7% more accurate on AP
these researchers compare their 7M parameter model to the ultralytics 1.4M parameters model and talk about how it works better. When in my opinion they should be comparing their YOLOv7-tiny-SiLU to the yolov5s not the yolov5n this is awful idk wtf is cvpr even. Comparing models like this to make your model seem SO GOOD is just so **dishonest** to me i cba to read papers like this

+ me and my colleague tried using the yolov7 code on our VERY SIMPLE 1(him 6) class object detection problems that are definitey solvable with the yolo algorithm and  the yolov7 results were significantly worse than the yolov5(v6.0) that was on ultralytics repo at the time.
+ yolov7 repo is horribly maintained with many good PRs not being accepted (one of those was mine >:( )
